cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -536,10 +573,31 @@ boolean tryWrite() {
     }
 
     /**
+     * Attempts to refresh the underlying FileWriter.
+     *
+     * @return true if become the writer leader and performed the refresh, false otherwise.
+     */
+    boolean refresh() throws IOException {
+      if (!writerFlag.compareAndSet(false, true)) {
+        return false;
+      }
+
+      try {
+        FileWriter fileWriter = this.getFileWriter();
+        if (fileWriter instanceof TimePartitionedStreamFileWriter) {
```

Comment:
That's not a good one. It's too specific. introduce an interface, like Refreshable.